src/Test/AppModel/DRT/Application/AppThreadingModel/DrtAppThreadingModel.cs

```diff
@@ -0,0 +1,359 @@
+        
+        private bool                _propertiesTestPassed;
+        private bool                _findResourceTestPassed;
+        private bool                _currentTestPassed;
+        private bool                _runTestPassed;
+        private bool                _shutdownTestPassed;
+        private bool                _shutdownmodeTestPassed;
+        private bool                _windowsTestPassed;
+        private bool                _mainwindowTestPassed;
+        private bool                _win2ResourceInvalidationPassed;
+        private bool                _win1ResourceInvalidationPassed;
+            
+        private bool                _result = true;
+
+        // This DRTs does the following tests:
+        //
+        // 1) Verify that APIs that are specific to Application thread throw an exception
+        //    when called from a worker thread.  We do this test before calling 
+        //    Application.Run since we also want to verify run.
+        //
+        // 2) Add a properties to the Properties property on App thread and verify reading it from 
+        //    worker thread.
+        //
+        // 3) Add a resource on App thread and verify Finding (FindResource) it from worker thread.
+        //
+        // 4) Read Current from App and worker thread
+        //
+        // 5) Set up resource references for Window property on two windows created on different
+        //    threads.  Then, update resources from worker thread and verify that resources are 
+        //    invalidated on both threads/windows.  Resources on the thread adding the resource
+        //    invalidated synchronously while worker thread resources are invalidated asynchronously.
+        //
+        [STAThread]
+        public static int Main()
+        {
+            _app = new DrtAppThreadingModel();
+            if (!TestAppThreadOnlyAPIs())
+            {
+                _app._logger.Log("Failed.");
+                return 1;
+            }
+            else
+            {
+                _app.Run();
+                if (_app._result)
+                {
+                    _app._logger.Log("Passed.");
+                    return 0;
+                }
+                else
+                {
+                    _app._logger.Log("Failed.");
+                    return 1;
+                }
+            }
+        }
+
+        private static bool TestAppThreadOnlyAPIs()
+        {
+            Thread t2 = new Thread(new ThreadStart(TestAppThreadOnlyAPIsImpl));
+            t2.SetApartmentState(ApartmentState.STA);
+            t2.Start();
+
+            // of t2 terminated
+            if (t2.Join(TimeSpan.FromMilliseconds(30000)) == true)
+            {
+                return VerifyAppThreadOnlyAPIs();
+            }
+            else
+            {
+                // t2 hung
+                // REPORT error
+                _app._logger.Log("Thread 2 hung inside TestAppThreadOnlyAPIs");
+                t2.Abort();
```

Comment:
Can we use the Abort method in .NET 5?

Reply:
@lindexi I looked at this. And it should fail for .NET 5. I will create an issue for the same and the fix for this can be taken then.